r/src/array_to_vector.cpp

```diff
@@ -656,8 +656,15 @@ class Converter_Null : public Converter {
     return Status::OK();
   }
 };
+
 std::shared_ptr<Converter> Converter::Make(const ArrayVector& arrays) {
-  switch (arrays[0]->type_id()) {
+  if (arrays.empty()) {
```

Comment:
To save us future pain, should we hunt for other places that assume >0 length vectors? grepping `[0]` in `r/src` hits a few places that look similarly suspect.

Reply:
I checked them all and found one bug, I'll add a test and the fix in the current PR.